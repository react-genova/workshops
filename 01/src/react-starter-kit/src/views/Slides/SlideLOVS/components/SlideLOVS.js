//@flow
import React from 'react';
import styled from 'styled-components';

import CardDeveloperTools from './CardDeveloperTools';
import CardLibraries from './CardLibraries';
import CardReact from './CardReact';
import CardRedux from './CardRedux';
import CardStudy from './CardStudy';
import CardTesting from './CardTesting';

const Container = styled.div`
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content:space-evenly;
    align-items: flex-start;
    align-content: center;
`;

const CardPad = styled.div`
    padding: 0.4em;
`;

type Props = {
    openURL:(url:string)=>void
};

const SlideLOVS = ({openURL}:Props) => (
    <Container>
        <CardPad><CardDeveloperTools onClick={openURL} /></CardPad>
        <CardPad><CardReact onClick={openURL} /></CardPad>
        <CardPad><CardTesting onClick={openURL} /></CardPad>
        <CardPad><CardRedux onClick={openURL} /></CardPad>
        <CardPad><CardLibraries onClick={openURL} /></CardPad>
        <CardPad><CardStudy onClick={openURL} /></CardPad>
    </Container>
);

export default SlideLOVS;