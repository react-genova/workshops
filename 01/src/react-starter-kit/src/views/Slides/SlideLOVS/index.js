//@flow
export {default} from './components/SlideLOVS';