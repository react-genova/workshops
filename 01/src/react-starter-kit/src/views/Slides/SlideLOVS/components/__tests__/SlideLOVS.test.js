//@flow
import React from 'react';
import { shallow } from 'enzyme';
import SlideLOVS from '../SlideLOVS';

describe('Slide LOVS', () => {

    it('snapshot check', () => {
        const openURL = (url:string)=>{url.toLocaleLowerCase();};
        const component = shallow(<SlideLOVS openURL={openURL} />);
        expect(component).toMatchSnapshot();
    });

});