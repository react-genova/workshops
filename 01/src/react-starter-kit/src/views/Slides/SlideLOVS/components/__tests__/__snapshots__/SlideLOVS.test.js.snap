// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide LOVS snapshot check 1`] = `
<styled.div>
  <styled.div>
    <CardDeveloperTools
      onClick={[Function]}
    />
  </styled.div>
  <styled.div>
    <CardReact
      onClick={[Function]}
    />
  </styled.div>
  <styled.div>
    <CardTesting
      onClick={[Function]}
    />
  </styled.div>
  <styled.div>
    <CardRedux
      onClick={[Function]}
    />
  </styled.div>
  <styled.div>
    <CardLibraries
      onClick={[Function]}
    />
  </styled.div>
  <styled.div>
    <CardStudy
      onClick={[Function]}
    />
  </styled.div>
</styled.div>
`;
